git-restore: add examples (#5101)
tldr-pages/tldr: pages/common/git-restore.md

# git restore

> Restore working tree files. Requires git version 2.23+.
> See also `git checkout` and `git reset`.
> More information: <https://git-scm.com/docs/git-restore/>.

- Restore an unstaged file to the version of the current commit (HEAD):

`git restore {{path/to/file}}`

- Restore an unstaged file to the version of a specific commit:

`git restore --source {{commit}} {{path/to/file}}`

- Discard all unstaged changes to tracked files:

> `git restore :/`

Reviewer: Love this change.  It's so effective! 😍

- Unstage a file:

`git restore --staged {{path/to/file}}`

- Unstage all files:

`git restore --staged :/`

> - Discard all changes to files, staged or not:

Reviewer 2: Perhaps this will read better?
```suggestion
- Discard all changes to files, both staged and unstaged:
```

Author: good point, that sounds better! 👍

`git restore --worktree --staged :/`

> - Interactively select parts of files to restore:

Reviewer 2: Does this allow the user to choose which files to restore, or which sections of files to restore?

Author: which sections of files. git calls those 'hunks', but i think 'parts' or 'sections' is easier to understand.

Author: it's similar to git add --patch

Reviewer: I agree.  How about changing it to 'sections' to make a bit clearer?
```suggestion
- Interactively select sections of files to restore:
```
